ui/src/shared/components/FunctionSelector.js

```diff
@@ -0,0 +1,82 @@
+import React, {Component, PropTypes} from 'react'
+import classnames from 'classnames'
+import _ from 'lodash'
+import {INFLUXQL_FUNCTIONS} from 'src/data_explorer/constants'
+
+class FunctionSelector extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      localSelectedItems: this.props.selectedItems,
+    }
+
+    this.onSelect = ::this.onSelect
+    this.onApplyFunctions = ::this.onApplyFunctions
+  }
+
+  onSelect(item, e) {
+    e.stopPropagation()
+
+    const {localSelectedItems} = this.state
+
+    let nextItems
+    if (this.isSelected(item)) {
+      nextItems = localSelectedItems.filter(i => i !== item)
+    } else {
+      nextItems = localSelectedItems.concat(item)
```

Comment:
after some thought, I'd prefer to have this syntax modernized:
`nextItems = [...localSelectedItems, item]`